compiler/rustc_middle/src/mir/query.rs

```diff
@@ -39,14 +39,14 @@ pub enum UnsafetyViolationDetails {
     AccessToUnionField,
     MutationOfLayoutConstrainedField,
     BorrowOfLayoutConstrainedField,
-    CallToFunctionWith,
+    CallToFunctionWith(DefId),
 }
 
 impl UnsafetyViolationDetails {
     pub fn description_and_note(&self) -> (&'static str, &'static str) {
```

Comment:
Change the first return value to a Cow and use the new def id field to report better diagnostics by using `tcx.def_path_str(def_id)` to report a better diagnostic,. You'll need to pass in a `TyCtxt` argument to this function for that

Reply:
The intent wasn't to change the error message, but to just provide the `def_id`. I've tried to use `def_path_str` here but I'm getting an ICE calling `tcx.def_path_str(def_id)` in this context.


<details>
<summary>Details (windows build)</summary>

```
❯ python3 ./x.py build compiler/rustc

Updating only changed submodules
  Submodules updated in 0.06 seconds
extracting G:\workspace\rust\build\cache\2022-04-05\rust-std-beta-x86_64-pc-windows-msvc.tar.xz
extracting G:\workspace\rust\build\cache\2022-04-05\rustc-beta-x86_64-pc-windows-msvc.tar.xz
extracting G:\workspace\rust\build\cache\2022-04-05\cargo-beta-x86_64-pc-windows-msvc.tar.xz
extracting G:\workspace\rust\build\cache\2022-04-05\rustfmt-nightly-x86_64-pc-windows-msvc.tar.xz
extracting G:\workspace\rust\build\cache\llvm-311e2683e1bad87715b1558f7900e294d24ce491-False\rust-dev-nightly-x86_64-pc-windows-msvc.tar.xz
Building rustbuild
   Compiling proc-macro2 v1.0.30
   Compiling winapi v0.3.9
   Compiling unicode-xid v0.2.2
   Compiling memchr v2.4.1
   Compiling syn v1.0.80
   Compiling cfg-if v1.0.0
   Compiling serde_derive v1.0.125
   Compiling lazy_static v1.4.0
   Compiling serde v1.0.125
   Compiling log v0.4.14
   Compiling regex-syntax v0.6.25
   Compiling ryu v1.0.5
   Compiling regex-automata v0.1.10
   Compiling crossbeam-utils v0.8.6
   Compiling fnv v1.0.7
   Compiling once_cell v1.7.2
   Compiling serde_json v1.0.59
   Compiling libc v0.2.121
   Compiling bootstrap v0.0.0 (G:\workspace\rust\src\bootstrap)
   Compiling itoa v0.4.6
   Compiling unicode-width v0.1.8
   Compiling cc v1.0.69
   Compiling getopts v0.2.21                                                                                                                                                                                   
   Compiling thread_local v1.1.4                                                                                                                                                                               
   Compiling quote v1.0.7
   Compiling cmake v0.1.44
   Compiling aho-corasick v0.7.18
   Compiling bstr v0.2.13
   Compiling regex v1.5.5
   Compiling globset v0.4.5
   Compiling winapi-util v0.1.5
   Compiling opener v0.5.0
   Compiling filetime v0.2.14
   Compiling same-file v1.0.6
   Compiling walkdir v2.3.2
   Compiling ignore v0.4.17
   Compiling toml v0.5.7
    Finished dev [unoptimized] target(s) in 24.70s
Building stage0 std artifacts (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc)
   Compiling compiler_builtins v0.1.71
   Compiling core v0.0.0 (G:\workspace\rust\library\core)
   Compiling libc v0.2.121
   Compiling cc v1.0.69
   Compiling memchr v2.4.1
   Compiling std v0.0.0 (G:\workspace\rust\library\std)
   Compiling unwind v0.0.0 (G:\workspace\rust\library\unwind)
   Compiling rustc-std-workspace-core v1.99.0 (G:\workspace\rust\library\rustc-std-workspace-core)
   Compiling alloc v0.0.0 (G:\workspace\rust\library\alloc)
   Compiling cfg-if v0.1.10
   Compiling adler v0.2.3
   Compiling rustc-demangle v0.1.21
   Compiling rustc-std-workspace-alloc v1.99.0 (G:\workspace\rust\library\rustc-std-workspace-alloc)
   Compiling panic_unwind v0.0.0 (G:\workspace\rust\library\panic_unwind)
   Compiling panic_abort v0.0.0 (G:\workspace\rust\library\panic_abort)
   Compiling gimli v0.25.0
   Compiling miniz_oxide v0.4.0
   Compiling object v0.26.2
   Compiling std_detect v0.1.5 (G:\workspace\rust\library\stdarch\crates\std_detect)
   Compiling hashbrown v0.12.0
   Compiling addr2line v0.16.0
   Compiling rustc-std-workspace-std v1.99.0 (G:\workspace\rust\library\rustc-std-workspace-std)
   Compiling proc_macro v0.0.0 (G:\workspace\rust\library\proc_macro)
   Compiling unicode-width v0.1.8
   Compiling getopts v0.2.21
   Compiling test v0.0.0 (G:\workspace\rust\library\test)
    Finished release [optimized] target(s) in 40.67s
Copying stage0 std from stage0 (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc / x86_64-pc-windows-msvc)
Building stage0 compiler artifacts (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc)
   Compiling proc-macro2 v1.0.30
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.80
   Compiling version_check v0.9.3
   Compiling cfg-if v1.0.0
   Compiling winapi v0.3.9
   Compiling cfg-if v0.1.10
   Compiling getrandom v0.2.0
   Compiling once_cell v1.7.2
   Compiling smallvec v1.7.0
   Compiling lazy_static v1.4.0
   Compiling autocfg v1.1.0
   Compiling log v0.4.14
   Compiling cc v1.0.69
   Compiling parking_lot_core v0.8.5
   Compiling scopeguard v1.1.0
   Compiling pin-project-lite v0.2.8
   Compiling libc v0.2.121
   Compiling rustc-hash v1.1.0
   Compiling typenum v1.12.0
   Compiling bitflags v1.2.1
   Compiling ppv-lite86 v0.2.8
   Compiling stable_deref_trait v1.2.0
   Compiling proc-macro-hack v0.5.19
   Compiling arrayvec v0.7.0                                                                                                                                                                                   
   Compiling cpufeatures v0.2.1                                                                                                                                                                                
   Compiling tinystr v0.3.4
   Compiling jobserver v0.1.24
   Compiling rustc_graphviz v0.0.0 (G:\workspace\rust\compiler\rustc_graphviz)
   Compiling memmap2 v0.2.1
   Compiling unicode-width v0.1.8
   Compiling scoped-tls v1.0.0
   Compiling unic-common v0.9.0
   Compiling unic-char-range v0.9.0
   Compiling self_cell v0.10.2
   Compiling annotate-snippets v0.8.0
   Compiling rustc_fs_util v0.0.0 (G:\workspace\rust\compiler\rustc_fs_util)
   Compiling serde_derive v1.0.125
   Compiling serde v1.0.125
   Compiling either v1.6.0
   Compiling datafrog v2.0.1
   Compiling memchr v2.4.1
   Compiling regex-syntax v0.6.25
   Compiling tinyvec v0.3.4
   Compiling crc32fast v1.2.0
   Compiling snap v1.0.1
   Compiling adler v0.2.3
   Compiling fixedbitset v0.2.0
   Compiling unicode-script v0.5.3
   Compiling fallible-iterator v0.2.0
   Compiling punycode v0.4.1
   Compiling rustc-demangle v0.1.21
   Compiling ryu v1.0.5
   Compiling serde_json v1.0.59
   Compiling pathdiff v0.2.1
   Compiling itoa v0.4.6
   Compiling rustc_error_codes v0.0.0 (G:\workspace\rust\compiler\rustc_error_codes)
   Compiling instant v0.1.12
   Compiling odht v0.3.1
   Compiling tracing-core v0.1.21
   Compiling sharded-slab v0.1.1
   Compiling lock_api v0.4.5
   Compiling type-map v0.4.0
   Compiling thread_local v1.1.4
   Compiling rustc_arena v0.0.0 (G:\workspace\rust\compiler\rustc_arena)
   Compiling unic-langid-impl v0.9.0
   Compiling getopts v0.2.21
   Compiling unic-ucd-version v0.9.0                                                                                                                                                                           
   Compiling unic-char-property v0.9.0                                                                                                                                                                         
   Compiling ahash v0.7.4                                                                                                                                                                                      
   Compiling indexmap v1.8.0
   Compiling generic-array v0.14.4                                                                                                                                                                             
   Compiling itertools v0.10.1
   Compiling miniz_oxide v0.4.0
   Compiling unicode-normalization v0.1.13
   Compiling unic-emoji-char v0.9.0
   Compiling rustc_apfloat v0.0.0 (G:\workspace\rust\compiler\rustc_apfloat)                                                                                                                                   
   Compiling ena v0.14.0                                                                                                                                                                                       
   Compiling polonius-engine v0.13.0                                                                                                                                                                           
   Compiling tracing-log v0.1.2                                                                                                                                                                                
   Compiling rand_core v0.6.2                                                                                                                                                                                  
   Compiling rustc_lexer v0.1.0 (G:\workspace\rust\compiler\rustc_lexer)                                                                                                                                       
   Compiling flate2 v1.0.16                                                                                                                                                                                    
   Compiling quote v1.0.7                                                                                                                                                                                      
   Compiling aho-corasick v0.7.18                                                                                                                                                                              
   Compiling rand_chacha v0.3.0
   Compiling rand_xoshiro v0.6.0
   Compiling unicode-security v0.0.5
   Compiling psm v0.1.16
   Compiling stacker v0.1.14
   Compiling rustc_llvm v0.0.0 (G:\workspace\rust\compiler\rustc_llvm)
   Compiling hashbrown v0.11.2
   Compiling rand v0.8.4
   Compiling regex-automata v0.1.10
   Compiling cstr v0.2.8
   Compiling regex v1.5.5
   Compiling crypto-common v0.1.2                                                                                                                                                                              
   Compiling block-buffer v0.10.2                                                                                                                                                                              
   Compiling matchers v0.1.0
   Compiling digest v0.10.2
   Compiling rustc_serialize v0.0.0 (G:\workspace\rust\compiler\rustc_serialize)                                                                                                                               
   Compiling petgraph v0.5.1
   Compiling gimli v0.26.1
   Compiling object v0.28.1
   Compiling sha-1 v0.10.0
   Compiling sha2 v0.10.1
   Compiling md-5 v0.10.0                                                                                                                                                                                      
   Compiling remove_dir_all v0.5.3
   Compiling atty v0.2.14
   Compiling winapi-util v0.1.5
   Compiling termize v0.1.1
   Compiling ansi_term v0.12.1                                                                                                                                                                                 
   Compiling libloading v0.7.1                                                                                                                                                                                 
   Compiling parking_lot v0.11.2                                                                                                                                                                               
   Compiling tempfile v3.2.0
   Compiling termcolor v1.1.2
   Compiling measureme v10.0.0
   Compiling synstructure v0.12.6
   Compiling tracing-attributes v0.1.18
   Compiling rustc_macros v0.1.0 (G:\workspace\rust\compiler\rustc_macros)
   Compiling unic-langid-macros-impl v0.9.0
   Compiling thiserror-impl v1.0.30
   Compiling chalk-derive v0.80.0
   Compiling unic-langid-macros v0.9.0
   Compiling chalk-ir v0.80.0
   Compiling unic-langid v0.9.0
   Compiling intl-memoizer v0.5.1
   Compiling fluent-langneg v0.13.0
   Compiling intl_pluralrules v7.0.1
   Compiling tracing v0.1.29                                                                                                                                                                                   
   Compiling thiserror v1.0.30
   Compiling tracing-subscriber v0.3.3
   Compiling thorin-dwp v0.2.0
   Compiling fluent-syntax v0.11.0
   Compiling fluent-bundle v0.15.2
   Compiling rustc_index v0.0.0 (G:\workspace\rust\compiler\rustc_index)
   Compiling rustc_data_structures v0.0.0 (G:\workspace\rust\compiler\rustc_data_structures)
   Compiling tracing-tree v0.2.0
   Compiling chalk-solve v0.80.0
   Compiling rustc_log v0.0.0 (G:\workspace\rust\compiler\rustc_log)
   Compiling rustc_span v0.0.0 (G:\workspace\rust\compiler\rustc_span)
   Compiling rustc_type_ir v0.0.0 (G:\workspace\rust\compiler\rustc_type_ir)
   Compiling chalk-engine v0.80.0
   Compiling rustc_ast v0.0.0 (G:\workspace\rust\compiler\rustc_ast)
   Compiling rustc_feature v0.0.0 (G:\workspace\rust\compiler\rustc_feature)
   Compiling rustc_error_messages v0.0.0 (G:\workspace\rust\compiler\rustc_error_messages)
   Compiling rustc_target v0.0.0 (G:\workspace\rust\compiler\rustc_target)
   Compiling rustc_parse_format v0.0.0 (G:\workspace\rust\compiler\rustc_parse_format)
   Compiling gsgdt v0.1.2
   Compiling rls-span v0.5.3
   Compiling rls-data v0.19.1
   Compiling rustc_ast_pretty v0.0.0 (G:\workspace\rust\compiler\rustc_ast_pretty)
   Compiling rustc_hir v0.0.0 (G:\workspace\rust\compiler\rustc_hir)
   Compiling rustc_lint_defs v0.0.0 (G:\workspace\rust\compiler\rustc_lint_defs)
   Compiling rustc_hir_pretty v0.0.0 (G:\workspace\rust\compiler\rustc_hir_pretty)
   Compiling rustc_errors v0.0.0 (G:\workspace\rust\compiler\rustc_errors)
   Compiling rustc_session v0.0.0 (G:\workspace\rust\compiler\rustc_session)
   Compiling rustc_attr v0.0.0 (G:\workspace\rust\compiler\rustc_attr)
   Compiling rustc_query_system v0.0.0 (G:\workspace\rust\compiler\rustc_query_system)
   Compiling rustc_parse v0.0.0 (G:\workspace\rust\compiler\rustc_parse)
   Compiling rustc_middle v0.0.0 (G:\workspace\rust\compiler\rustc_middle)
   Compiling rustc_ast_lowering v0.0.0 (G:\workspace\rust\compiler\rustc_ast_lowering)
   Compiling rustc_ast_passes v0.0.0 (G:\workspace\rust\compiler\rustc_ast_passes)
   Compiling rustc_expand v0.0.0 (G:\workspace\rust\compiler\rustc_expand)
   Compiling rustc_builtin_macros v0.0.0 (G:\workspace\rust\compiler\rustc_builtin_macros)
   Compiling rustc_infer v0.0.0 (G:\workspace\rust\compiler\rustc_infer)
   Compiling rustc_metadata v0.0.0 (G:\workspace\rust\compiler\rustc_metadata)
   Compiling rustc_mir_dataflow v0.0.0 (G:\workspace\rust\compiler\rustc_mir_dataflow)
   Compiling rustc_incremental v0.0.0 (G:\workspace\rust\compiler\rustc_incremental)
   Compiling rustc_symbol_mangling v0.0.0 (G:\workspace\rust\compiler\rustc_symbol_mangling)
   Compiling rustc_query_impl v0.0.0 (G:\workspace\rust\compiler\rustc_query_impl)
   Compiling rustc_monomorphize v0.0.0 (G:\workspace\rust\compiler\rustc_monomorphize)
   Compiling rustc_passes v0.0.0 (G:\workspace\rust\compiler\rustc_passes)
   Compiling rustc_save_analysis v0.0.0 (G:\workspace\rust\compiler\rustc_save_analysis)
   Compiling rustc_codegen_ssa v0.0.0 (G:\workspace\rust\compiler\rustc_codegen_ssa)                                                                                                                           
   Compiling rustc_resolve v0.0.0 (G:\workspace\rust\compiler\rustc_resolve)
   Compiling rustc_trait_selection v0.0.0 (G:\workspace\rust\compiler\rustc_trait_selection)
   Compiling rustc_codegen_llvm v0.0.0 (G:\workspace\rust\compiler\rustc_codegen_llvm)
   Compiling rustc_lint v0.0.0 (G:\workspace\rust\compiler\rustc_lint)                                                                                                                                         
   Compiling rustc_const_eval v0.0.0 (G:\workspace\rust\compiler\rustc_const_eval)
   Compiling rustc_ty_utils v0.0.0 (G:\workspace\rust\compiler\rustc_ty_utils)                                                                                                                                 
   Compiling rustc_traits v0.0.0 (G:\workspace\rust\compiler\rustc_traits)
   Compiling rustc_mir_build v0.0.0 (G:\workspace\rust\compiler\rustc_mir_build)
   Compiling rustc_typeck v0.0.0 (G:\workspace\rust\compiler\rustc_typeck)                                                                                                                                     
   Compiling rustc_plugin_impl v0.0.0 (G:\workspace\rust\compiler\rustc_plugin_impl)
   Compiling rustc_mir_transform v0.0.0 (G:\workspace\rust\compiler\rustc_mir_transform)                                                                                                                       
   Compiling rustc_borrowck v0.0.0 (G:\workspace\rust\compiler\rustc_borrowck)                                                                                                                                 
   Compiling rustc_privacy v0.0.0 (G:\workspace\rust\compiler\rustc_privacy)
   Compiling rustc_interface v0.0.0 (G:\workspace\rust\compiler\rustc_interface)
   Compiling rustc_driver v0.0.0 (G:\workspace\rust\compiler\rustc_driver)                                                                                                                                     
   Compiling rustc-main v0.0.0 (G:\workspace\rust\compiler\rustc)
    Finished release [optimized] target(s) in 7m 43s
Copying stage0 rustc from stage0 (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc / x86_64-pc-windows-msvc)
Assembling stage1 compiler (x86_64-pc-windows-msvc)
Building stage1 std artifacts (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc)
   Compiling compiler_builtins v0.1.71
   Compiling core v0.0.0 (G:\workspace\rust\library\core)
   Compiling libc v0.2.121
   Compiling cc v1.0.69
   Compiling memchr v2.4.1
   Compiling std v0.0.0 (G:\workspace\rust\library\std)
   Compiling unwind v0.0.0 (G:\workspace\rust\library\unwind)
   Compiling rustc-std-workspace-core v1.99.0 (G:\workspace\rust\library\rustc-std-workspace-core)
   Compiling alloc v0.0.0 (G:\workspace\rust\library\alloc)
   Compiling cfg-if v0.1.10
   Compiling adler v0.2.3
   Compiling rustc-demangle v0.1.21
error: internal compiler error: no warnings or errors encountered even though `delayed_good_path_bugs` issued

error: internal compiler error: trimmed_def_paths constructed
  |
  = note: delayed at    0: std::backtrace_rs::backtrace::trace_unsynchronized
             1: std::backtrace::Backtrace::disabled
             2: std::backtrace::Backtrace::force_capture
             3: <rustc_errors::Handler>::delay_good_path_bug::<&str>
             4: rustc_middle::ty::print::pretty::trimmed_def_paths
             5: rustc_query_system::query::plumbing::force_query::<rustc_query_impl::queries::trait_def, rustc_query_impl::plumbing::QueryCtxt>
             6: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::trimmed_def_paths, rustc_query_impl::plumbing::QueryCtxt>
             7: <rustc_middle::ty::print::pretty::FmtPrinter as rustc_middle::ty::print::pretty::PrettyPrinter>::try_print_trimmed_def_path
             8: <rustc_middle::ty::print::pretty::FmtPrinter as rustc_middle::ty::print::Printer>::print_def_path
             9: <rustc_middle::ty::context::TyCtxt>::def_path_str_with_substs
            10: <rustc_middle::mir::query::UnsafetyViolationDetails>::description_and_note
            11: rustc_mir_transform::check_unsafety::check_unsafety
            12: <rustc_session::session::Session>::time::<(), rustc_interface::passes::analysis::{closure#3}>
            13: rustc_interface::passes::analysis
            14: rustc_query_system::query::plumbing::force_query::<rustc_query_impl::queries::trait_def, rustc_query_impl::plumbing::QueryCtxt>
            15: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::analysis, rustc_query_impl::plumbing::QueryCtxt>
            16: <rustc_interface::passes::QueryContext>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}::{closure#3}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
            17: RINvMs2_NtCsgyiwMCGpMzM_15rustc_interface7queriesNtNtB8_9interface8Compiler5enterNCNCNvCsgHaxX4ZEAvI_12rustc_driver12run_compilers_0s0_0INtNtCs5ovHIRWrShO_4core6result6ResultINtNtB2f_6option6OptionNtB6_6LinkerENtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuarante
            18: RINvCsbP1H8PvM7wb_10rustc_span15with_source_mapINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCINvNtCsgyiwMCGpMzM_15rustc_interface9interface23create_compiler_and_runBJ_NCNvCsgHaxX4ZEAvI_12rustc_driver12run_compile
            19: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
            20: RINvMs_Csh6pxvok7FZi_10scoped_tlsINtB5_9ScopedKeyNtCsbP1H8PvM7wb_10rustc_span14SessionGlobalsE3setNCINvNtCsgyiwMCGpMzM_15rustc_interface9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCNvCsgH
            21: RINvNtNtCs9kCJ5EiIH6k_3std10sys_common9backtrace28___rust_begin_short_backtraceNCINvNtCsgyiwMCGpMzM_15rustc_interface4util31run_in_thread_pool_with_globalsNCINvNtB1m_9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_er
            22: RINvNtCs9kCJ5EiIH6k_3std9panicking3tryINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedEINtNtNtBF_5panic11unwind_safe16AssertUnwindSafeNCNCINvMNtB4_6threadNtB2S_7Builder16spawn_unchecked_NCINvNtCsgyiwMCGpMzM_15rustc_in
            23: <std::thread::Builder>::spawn_unchecked::<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver::run_compiler::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>
            24: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
            25: std::sys::windows::thread::Thread::new
            26: BaseThreadInitThunk
            27: RtlUserThreadStart


thread 'rustc' panicked at 'Box<dyn Any>', compiler\rustc_errors\src\lib.rs:1347:13
stack backtrace:
   0: std::panicking::begin_panic::<rustc_errors::ExplicitBug>
   1: RINvYTNtNtCsbP1H8PvM7wb_10rustc_span13span_encoding4SpanNtCsjdlliSgPPGe_20rustc_error_messages17DiagnosticMessageENtNtCs5ovHIRWrShO_4core4hash4Hash10hash_sliceNtNtCsjfStFDLtnAf_21rustc_data_structures13stable_hasher12StableHasherECs6hk4i3bhPrJ_12rustc_err
   2: <rustc_errors::HandlerInner>::flush_delayed::<core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<rustc_errors::DelayedDiagnostic>, <rustc_errors::DelayedDiagnostic>::decorate>, &str>       
   3: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
   4: core::ptr::drop_in_place::<rustc_errors::Handler>
   5: core::ptr::drop_in_place::<rustc_session::cgu_reuse_tracker::CguReuseTracker>
   6: core::ptr::drop_in_place::<alloc::rc::Rc<rustc_session::session::Session>>
   7: core::ptr::drop_in_place::<rustc_interface::interface::Compiler>
   8: RINvCsbP1H8PvM7wb_10rustc_span15with_source_mapINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCINvNtCsgyiwMCGpMzM_15rustc_interface9interface23create_compiler_and_runBJ_NCNvCsgHaxX4ZEAvI_12rustc_driver12run_compile
   9: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
  10: RINvMs_Csh6pxvok7FZi_10scoped_tlsINtB5_9ScopedKeyNtCsbP1H8PvM7wb_10rustc_span14SessionGlobalsE3setNCINvNtCsgyiwMCGpMzM_15rustc_interface9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCNvCsgH
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.0-dev running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=10000 -C debuginfo=0 -C symbol-mangling-version=legacy -Z unstable-options -Z unstable-options -Z macro-backtrace -Z unstable-options -C split-debuginfo=packed -C target-feature=+crt-static -C prefer-dynamic -C llvm-args=-import-instr-limit=10 -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -C panic=abort -Z force-unstable-if-unmarked

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: internal compiler error: trimmed_def_paths constructed
  |
  = note: delayed at    0: std::backtrace_rs::backtrace::trace_unsynchronized
             1: std::backtrace::Backtrace::disabled
             2: std::backtrace::Backtrace::force_capture
             3: <rustc_errors::Handler>::delay_good_path_bug::<&str>
             4: rustc_middle::ty::print::pretty::trimmed_def_paths
             5: RINvXNtCslbknMW6Bx95_12rustc_middle9dep_graphNtNtB3_8dep_node7DepKindNtNtCsclfTmamBV9V_18rustc_query_system9dep_graph7DepKind9with_depsNCINvMs1_NtB17_5graphINtB2i_8DepGraphBH_E14with_task_implNtNtNtB5_2ty7context6TyCtxtuINtNtCs5ovHIRWrShO_4core6option6Opt
             6: RINvMs1_NtNtCsclfTmamBV9V_18rustc_query_system9dep_graph5graphINtB6_8DepGraphNtNtNtCslbknMW6Bx95_12rustc_middle9dep_graph8dep_node7DepKindE9with_taskNtNtNtB1j_2ty7context6TyCtxtuINtNtNtNtCs9kCJ5EiIH6k_3std11collections4hash3map7HashMapNtNtCsbP1H8PvM7wb_10
             7: rustc_query_system::query::plumbing::force_query::<rustc_query_impl::queries::trait_def, rustc_query_impl::plumbing::QueryCtxt>
             8: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::trimmed_def_paths, rustc_query_impl::plumbing::QueryCtxt>
             9: <rustc_middle::ty::print::pretty::FmtPrinter as rustc_middle::ty::print::pretty::PrettyPrinter>::try_print_trimmed_def_path
            10: <rustc_middle::ty::print::pretty::FmtPrinter as rustc_middle::ty::print::Printer>::print_def_path
            11: <rustc_middle::ty::context::TyCtxt>::def_path_str_with_substs
            12: <rustc_middle::mir::query::UnsafetyViolationDetails>::description_and_note
            13: rustc_mir_transform::check_unsafety::check_unsafety
            14: <rustc_session::session::Session>::time::<(), rustc_interface::passes::analysis::{closure#3}>
            15: rustc_interface::passes::analysis
            16: RINvXNtCslbknMW6Bx95_12rustc_middle9dep_graphNtNtB3_8dep_node7DepKindNtNtCsclfTmamBV9V_18rustc_query_system9dep_graph7DepKind9with_depsNCINvMs1_NtB17_5graphINtB2i_8DepGraphBH_E14with_task_implNtNtNtB5_2ty7context6TyCtxtuINtNtCs5ovHIRWrShO_4core6option6Opt
            17: RINvMs1_NtNtCsclfTmamBV9V_18rustc_query_system9dep_graph5graphINtB6_8DepGraphNtNtNtCslbknMW6Bx95_12rustc_middle9dep_graph8dep_node7DepKindE9with_taskNtNtNtB1j_2ty7context6TyCtxtuINtNtCs5ovHIRWrShO_4core6option6OptionNtNtNtCs22koZPj0VQy_9rustc_ast6expand9a
            18: rustc_query_system::query::plumbing::force_query::<rustc_query_impl::queries::trait_def, rustc_query_impl::plumbing::QueryCtxt>
            19: rustc_query_system::query::plumbing::get_query::<rustc_query_impl::queries::analysis, rustc_query_impl::plumbing::QueryCtxt>
            20: <rustc_interface::passes::QueryContext>::enter::<rustc_driver::run_compiler::{closure#1}::{closure#2}::{closure#3}, core::result::Result<(), rustc_errors::ErrorGuaranteed>>
            21: RINvMs2_NtCsgyiwMCGpMzM_15rustc_interface7queriesNtNtB8_9interface8Compiler5enterNCNCNvCsgHaxX4ZEAvI_12rustc_driver12run_compilers_0s0_0INtNtCs5ovHIRWrShO_4core6result6ResultINtNtB2f_6option6OptionNtB6_6LinkerENtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuarante
            22: RINvCsbP1H8PvM7wb_10rustc_span15with_source_mapINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCINvNtCsgyiwMCGpMzM_15rustc_interface9interface23create_compiler_and_runBJ_NCNvCsgHaxX4ZEAvI_12rustc_driver12run_compile
            23: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
            24: RINvMs_Csh6pxvok7FZi_10scoped_tlsINtB5_9ScopedKeyNtCsbP1H8PvM7wb_10rustc_span14SessionGlobalsE3setNCINvNtCsgyiwMCGpMzM_15rustc_interface9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCNvCsgH
            25: RINvNtNtCs9kCJ5EiIH6k_3std10sys_common9backtrace28___rust_begin_short_backtraceNCINvNtCsgyiwMCGpMzM_15rustc_interface4util31run_in_thread_pool_with_globalsNCINvNtB1m_9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_er
            26: RINvNtCs9kCJ5EiIH6k_3std9panicking3tryINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedEINtNtNtBF_5panic11unwind_safe16AssertUnwindSafeNCNCINvMNtB4_6threadNtB2S_7Builder16spawn_unchecked_NCINvNtCsgyiwMCGpMzM_15rustc_in
            27: <std::thread::Builder>::spawn_unchecked::<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver::run_compiler::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>
            28: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
            29: std::sys::windows::thread::Thread::new
            30: BaseThreadInitThunk
            31: RtlUserThreadStart


thread 'rustc' panicked at 'Box<dyn Any>', compiler\rustc_errors\src\lib.rs:1347:13
stack backtrace:
   0: std::panicking::begin_panic::<rustc_errors::ExplicitBug>
   1: RINvYTNtNtCsbP1H8PvM7wb_10rustc_span13span_encoding4SpanNtCsjdlliSgPPGe_20rustc_error_messages17DiagnosticMessageENtNtCs5ovHIRWrShO_4core4hash4Hash10hash_sliceNtNtCsjfStFDLtnAf_21rustc_data_structures13stable_hasher12StableHasherECs6hk4i3bhPrJ_12rustc_err
   2: <rustc_errors::HandlerInner>::flush_delayed::<core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<rustc_errors::DelayedDiagnostic>, <rustc_errors::DelayedDiagnostic>::decorate>, &str>       
   3: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
   4: core::ptr::drop_in_place::<rustc_errors::Handler>
   5: core::ptr::drop_in_place::<rustc_session::cgu_reuse_tracker::CguReuseTracker>
   6: core::ptr::drop_in_place::<alloc::rc::Rc<rustc_session::session::Session>>
   7: core::ptr::drop_in_place::<rustc_interface::interface::Compiler>
   8: RINvCsbP1H8PvM7wb_10rustc_span15with_source_mapINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCINvNtCsgyiwMCGpMzM_15rustc_interface9interface23create_compiler_and_runBJ_NCNvCsgHaxX4ZEAvI_12rustc_driver12run_compile
   9: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
  10: RINvMs_Csh6pxvok7FZi_10scoped_tlsINtB5_9ScopedKeyNtCsbP1H8PvM7wb_10rustc_span14SessionGlobalsE3setNCINvNtCsgyiwMCGpMzM_15rustc_interface9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCNvCsgH
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.0-dev running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C incremental -C symbol-mangling-version=legacy -Z unstable-options -Z unstable-options -Z macro-backtrace -Z unstable-options -C split-debuginfo=packed -C target-feature=+crt-static -C prefer-dynamic -C llvm-args=-import-instr-limit=10 -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
thread 'rustc' panicked at 'Box<dyn Any>', compiler\rustc_errors\src\lib.rs:1347:13
stack backtrace:
error: could not compile `compiler_builtins`

Caused by:
  process didn't exit successfully: `G:\workspace\rust\build\bootstrap\debug\rustc --crate-name compiler_builtins C:\Users\Emil\.cargo\registry\src\github.com-1ecc6299db9ec823\compiler_builtins-0.1.71\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=10000 -C debuginfo=0 --cfg "feature=\"compiler-builtins\"" --cfg "feature=\"core\"" --cfg "feature=\"default\"" --cfg "feature=\"rustc-dep-of-std\"" -C metadata=879164927e46aba0 -C extra-filename=-879164927e46aba0 --out-dir G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps --target x86_64-pc-windows-msvc -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\release\deps --extern core=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps\librustc_std_workspace_core-710036e409e62f0d.rmeta --cap-lints allow -Csymbol-mangling-version=legacy -Zunstable-options -Zunstable-options --check-cfg=names() --check-cfg=values() --check-cfg=values(bootstrap) --check-cfg=values(stdarch_intel_sde) --check-cfg=values(no_fp_fmt_parse) --check-cfg=values(no_global_oom_handling) --check-cfg=values(freebsd12) --check-cfg=values(backtrace_in_libstd) "--check-cfg=values(target_env,\"libnx\")" "--check-cfg=values(target_os,\"watchos\")" "--check-cfg=values(target_arch,\"asmjs\",\"spirv\",\"nvptx\",\"nvptx64\",\"le32\",\"xtensa\")" --check-cfg=values(dont_compile_me) -Zmacro-backtrace -Zunstable-options -Csplit-debuginfo=packed -Ctarget-feature=+crt-static -Cprefer-dynamic -Cllvm-args=-import-instr-limit=10 -Cembed-bitcode=yes "-Zcrate-attr=doc(html_root_url=\"https://doc.rust-lang.org/nightly/\")" -Z binary-dep-depinfo --cfg "feature=\"unstable\"" --cfg "feature=\"mem-unaligned\""` (exit code: 0x80000003)
warning: build failed, waiting for other jobs to finish...
   0: std::panicking::begin_panic::<rustc_errors::ExplicitBug>
   1: RINvYTNtNtCsbP1H8PvM7wb_10rustc_span13span_encoding4SpanNtCsjdlliSgPPGe_20rustc_error_messages17DiagnosticMessageENtNtCs5ovHIRWrShO_4core4hash4Hash10hash_sliceNtNtCsjfStFDLtnAf_21rustc_data_structures13stable_hasher12StableHasherECs6hk4i3bhPrJ_12rustc_err
   2: <rustc_errors::HandlerInner>::flush_delayed::<core::iter::adapters::map::Map<alloc::vec::into_iter::IntoIter<rustc_errors::DelayedDiagnostic>, <rustc_errors::DelayedDiagnostic>::decorate>, &str>      
   3: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
   4: core::ptr::drop_in_place::<rustc_errors::Handler>
   5: core::ptr::drop_in_place::<rustc_session::cgu_reuse_tracker::CguReuseTracker>
   6: core::ptr::drop_in_place::<alloc::rc::Rc<rustc_session::session::Session>>
   7: core::ptr::drop_in_place::<rustc_interface::interface::Compiler>
   8: RINvCsbP1H8PvM7wb_10rustc_span15with_source_mapINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCINvNtCsgyiwMCGpMzM_15rustc_interface9interface23create_compiler_and_runBJ_NCNvCsgHaxX4ZEAvI_12rustc_driver12run_compile
   9: rustc_interface::interface::create_compiler_and_run::<core::result::Result<(), rustc_errors::ErrorGuaranteed>, rustc_driver::run_compiler::{closure#1}>
  10: RINvMs_Csh6pxvok7FZi_10scoped_tlsINtB5_9ScopedKeyNtCsbP1H8PvM7wb_10rustc_span14SessionGlobalsE3setNCINvNtCsgyiwMCGpMzM_15rustc_interface9interface12run_compilerINtNtCs5ovHIRWrShO_4core6result6ResultuNtCs6hk4i3bhPrJ_12rustc_errors15ErrorGuaranteedENCNvCsgH
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.0-dev running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C symbol-mangling-version=legacy -Z unstable-options -Z unstable-options -Z macro-backtrace -Z unstable-options -C split-debuginfo=packed -C target-feature=+crt-static -C prefer-dynamic -C llvm-args=-import-instr-limit=10 -C embed-bitcode=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `core`

Caused by:
  process didn't exit successfully: `G:\workspace\rust\build\bootstrap\debug\rustc --crate-name core --edition=2021 library\core\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C metadata=a73f3dcab1ae4e39 -C extra-filename=-a73f3dcab1ae4e39 --out-dir G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps --target x86_64-pc-windows-msvc -C incremental=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\incremental -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\release\deps -Csymbol-mangling-version=legacy -Zunstable-options -Zunstable-options --check-cfg=names() --check-cfg=values() --check-cfg=values(bootstrap) --check-cfg=values(stdarch_intel_sde) --check-cfg=values(no_fp_fmt_parse) --check-cfg=values(no_global_oom_handling) --check-cfg=values(freebsd12) --check-cfg=values(backtrace_in_libstd) "--check-cfg=values(target_env,\"libnx\")" "--check-cfg=values(target_os,\"watchos\")" "--check-cfg=values(target_arch,\"asmjs\",\"spirv\",\"nvptx\",\"nvptx64\",\"le32\",\"xtensa\")" --check-cfg=values(dont_compile_me) -Zmacro-backtrace -Zunstable-options -Csplit-debuginfo=packed -Ctarget-feature=+crt-static -Cprefer-dynamic -Cllvm-args=-import-instr-limit=10 -Cembed-bitcode=yes "-Zcrate-attr=doc(html_root_url=\"https://doc.rust-lang.org/nightly/\")" -Z binary-dep-depinfo` (exit code: 0x80000003) 
error: could not compile `memchr`

Caused by:
  process didn't exit successfully: `G:\workspace\rust\build\bootstrap\debug\rustc --crate-name memchr --edition=2018 C:\Users\Emil\.cargo\registry\src\github.com-1ecc6299db9ec823\memchr-2.4.1\src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 --cfg "feature=\"compiler_builtins\"" --cfg "feature=\"core\"" --cfg "feature=\"rustc-dep-of-std\"" -C metadata=1039e19a69690565 -C extra-filename=-1039e19a69690565 --out-dir G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps --target x86_64-pc-windows-msvc -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps -L dependency=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\release\deps --extern compiler_builtins=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps\libcompiler_builtins-879164927e46aba0.rmeta --extern core=G:\workspace\rust\build\x86_64-pc-windows-msvc\stage1-std\x86_64-pc-windows-msvc\release\deps\librustc_std_workspace_core-710036e409e62f0d.rmeta --cap-lints allow -Csymbol-mangling-version=legacy -Zunstable-options -Zunstable-options --check-cfg=names() --check-cfg=values() --check-cfg=values(bootstrap) --check-cfg=values(stdarch_intel_sde) --check-cfg=values(no_fp_fmt_parse) --check-cfg=values(no_global_oom_handling) --check-cfg=values(freebsd12) --check-cfg=values(backtrace_in_libstd) "--check-cfg=values(target_env,\"libnx\")" "--check-cfg=values(target_os,\"watchos\")" "--check-cfg=values(target_arch,\"asmjs\",\"spirv\",\"nvptx\",\"nvptx64\",\"le32\",\"xtensa\")" --check-cfg=values(dont_compile_me) -Zmacro-backtrace -Zunstable-options -Csplit-debuginfo=packed -Ctarget-feature=+crt-static -Cprefer-dynamic -Cllvm-args=-import-instr-limit=10 -Cembed-bitcode=yes "-Zcrate-attr=doc(html_root_url=\"https://doc.rust-lang.org/nightly/\")" -Z binary-dep-depinfo --cfg memchr_runtime_simd --cfg memchr_runtime_sse2 --cfg memchr_runtime_sse42 --cfg memchr_runtime_avx` (exit code: 0x80000003)
Build completed unsuccessfully in 0:09:44
```

</details>

Reply:
The ICEis due to the `description_and_note` being called everytime something unsafe is happening, even when that unsafety is valid (e.g in valid unsafe block/function).

What I don't understand is how the diag emit in `check_unsafety` is not shown to the user all the time

Why is it done this way? Seems wasteful.

Reply:
You could move the call at https://github.com/rust-lang/rust/blob/8d8135f003b35c3e109d013b2bed9ee9496da615/compiler/rustc_mir_build/src/check_unsafety.rs#L73 into the two match arms below that actually use it.

Reply:
> The intent wasn't to change the error message, but to just provide the `def_id`.

I'm aware of that, but the PR standing on its own inside rustc is better than having just a comment "this is used in external project X at https://some.url".

If we get a diagnostics improvement as a side effect, all the better

Reply:
> If we get a diagnostics improvement as a side effect, all the better

Sounds good! :D

> You could move the call at
> 
> https://github.com/rust-lang/rust/blob/8d8135f003b35c3e109d013b2bed9ee9496da615/compiler/rustc_mir_build/src/check_unsafety.rs#L73
> into the two match arms below that actually use it.

Ah, yeah [I thought](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/ICE.20in.20stage.201.20build.20of.20pr.20.2396294/near/279869133) this was the reason too, but it's not, that's `UnsafeOpKind::description_and_note`, the only time we call `UnsafetyViolationDetails::description_and_note` is in [`rustc_mir_transform::check_unsafety::check_unsafety`](https://github.com/Emilgardis/rust/blob/9893ee4c363cff47bc3b1dcd4af099ac16fc3fdf/compiler/rustc_interface/src/passes.rs#L961), and that function is only called [here](https://github.com/Emilgardis/rust/blob/9893ee4c363cff47bc3b1dcd4af099ac16fc3fdf/compiler/rustc_interface/src/passes.rs#L961) in `MIR_effect_checking`

made a [branch](https://github.com/rust-lang/rust/commit/9893ee4c363cff47bc3b1dcd4af099ac16fc3fdf) of the changes with the ICE, can push to this branch if you want so the ci can run through it, I'm unable to bless the changes to the tests though

Reply:
Oh... yea so the problem is that the lint could be allowed.

Maybe we need to take a step back and figure out why `def_path_str` is failing at all. Looking again at your backtrace now

Reply:
Aha!

https://github.com/rust-lang/rust/blob/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/compiler/rustc_middle/src/ty/print/pretty.rs#L2756

If you wrap the call with a with_no_trimmed_paths call (look at other calls to that helper), then the ICE will go away

Reply:
that worked, thank you!

It's a bit unfortunate as I'd really like it if it did show the trimmed path, but I guess it doesn't matter too much